editoast/editoast_models/src/rolling_stock/rolling_stock_category.rs

```diff
@@ -0,0 +1,64 @@
+use std::io::Write;
+use std::str::FromStr;
+
+use diesel::deserialize::FromSql;
+use diesel::deserialize::FromSqlRow;
+use diesel::expression::AsExpression;
+use diesel::pg::Pg;
+use diesel::pg::PgValue;
+use diesel::serialize::Output;
+use diesel::serialize::ToSql;
+use serde::Deserialize;
+use serde::Serialize;
+use strum::Display;
+use strum::EnumString;
+use utoipa::ToSchema;
+
+editoast_common::schemas! {
+    RollingStockCategory,
+}
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Serialize,
+    Deserialize,
+    ToSchema,
+    FromSqlRow,
+    AsExpression,
+    EnumString,
+    Display,
+)]
+#[diesel(sql_type = crate::tables::sql_types::RollingStockCategory)]
+#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum RollingStockCategory {
+    Unknown,
+    HighSpeedTrain,
+    IntercityTrain,
+    RegionalTrainMU,
+    NightTrain,
+    CommuterTrain,
+    FreightTrain,
+    FastFreightTrain,
+    TramTrain,
+    TouristicTrain,
+    WorkTrain,
+}
+
+impl FromSql<crate::tables::sql_types::RollingStockCategory, Pg> for RollingStockCategory {
+    fn from_sql(value: PgValue) -> diesel::deserialize::Result<Self> {
+        let s = std::str::from_utf8(value.as_bytes()).map_err(|_| "Invalid UTF-8 data")?;
+        RollingStockCategory::from_str(s)
+            .map_err(|_| "Unrecognized enum variant for RollingStockCategory".into())
+    }
+}
+
+impl ToSql<crate::tables::sql_types::RollingStockCategory, Pg> for RollingStockCategory {
+    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Pg>) -> diesel::serialize::Result {
+        let variant = self.to_string();
```

Comment:
`.to_string()` does allocate memory, but we might be able to avoid that by implementing `strum::IntoStaticStr` and then do
```suggestion
        let variant: &str = self.into();
```

Reply:
done.